test/check-argo-status.sh

```diff
@@ -30,29 +30,36 @@ do
   echo $WORKFLOW_STATUS | grep ${WORKFLOW_COMPLETE_KEYWORD} && s=0 && break || s=$? && printf "Workflow ${ARGO_WORKFLOW} is not finished.\n${WORKFLOW_STATUS}\nSleep for 20 seconds...\n" && sleep 20
 done
 
-# Check whether the argo workflow finished or not and exit if not.
+if [[ $s == 0 ]]; then
+  echo "Argo workflow finished."
+  if [[ ! -z "$TEST_RESULT_FOLDER" ]]; then
+    echo "Copy test result"
+    mkdir -p $ARTIFACT_DIR
+    gsutil cp -r "${TEST_RESULTS_GCS_DIR}"/* "${ARTIFACT_DIR}" || true
+  fi
+  argo get ${ARGO_WORKFLOW} -n ${NAMESPACE}
+  exit 0
+fi
+
+# Handling failed workflow
 if [[ $s != 0 ]]; then
- echo "Prow job Failed: Argo workflow timeout.."
- argo logs -w ${ARGO_WORKFLOW} -n ${NAMESPACE}
- exit $s
+  echo "Argo workflow timed out."
+else
+  echo "Argo workflow failed."
 fi
 
-echo "Argo workflow finished."
+echo "=========Argo Workflow Logs========="
+argo logs -w ${ARGO_WORKFLOW} -n ${NAMESPACE}
 
-if [[ ! -z "$TEST_RESULT_FOLDER" ]]
-then
-  echo "Copy test result"
-  mkdir -p $ARTIFACT_DIR
-  gsutil cp -r "${TEST_RESULTS_GCS_DIR}"/* "${ARTIFACT_DIR}" || true
-fi
+echo "=========Main workflow=============="
+argo get ${ARGO_WORKFLOW} -n ${NAMESPACE}
 
-if [[ $WORKFLOW_STATUS = *"${WORKFLOW_FAILED_KEYWORD}"* ]]; then
-  echo "Test workflow failed."
-  echo "=========Argo Workflow Logs========="
-  argo logs -w ${ARGO_WORKFLOW} -n ${NAMESPACE}
-  echo "===================================="
-  argo get ${ARGO_WORKFLOW} -n ${NAMESPACE}
-  exit 1
-else
-  argo get ${ARGO_WORKFLOW} -n ${NAMESPACE}
-fi
+echo "========Child workflows============="
+
+argo --namespace ${NAMESPACE} list --output=name |
```

Comment:
Is this going to print the main workflow for a second time?
Not a serious issue, you can also just rename "Child workflows" to "all workflows".

Reply:
Done.